web/src/users/add_url.rs

```diff
@@ -0,0 +1,32 @@
+use axum::{
+    extract::{Form, State},
+    response::{IntoResponse, Redirect},
+};
+use db::{urls::add_url, users::User};
+use serde::Deserialize;
+
+use crate::AppState;
+
+#[derive(Deserialize)]
+pub struct ArticleForm {
+    content: String,
+}
+
+#[axum::debug_handler(state = AppState)]
```

Comment:
This function is marked with `#[axum::debug_handler]` to enable debugging and state management.